recipes-app/node-red-gpio/files/0008-mraa-gpio-Add-support-for-IOT2050-SM.patch

```diff
@@ -0,0 +1,352 @@
+-            name: {value:""},
+-            pin:  {value:"", required: true, validate:RED.validators.number()},
+-            interval: {value:"100", required: true, validate:RED.validators.number()}
+-        },
+-        inputs:0,
+-        outputs:1,
+-        icon: "arrow.png",
+-        label: function() {
+-            return this.name||"A"+this.pin;
+-        },
+-        oneditprepare: function() {
+-            $('#node-input-pin').val(this.pin);
+-            $('#btype').text(RED.settings.mraaGpioAinBoardType);
+-            $('#ver-tip').text(RED.settings.mraaGpioAinMraaVersion);
+-        }
+-    });
++    if (RED.settings.mraaGpioAinBoardType !== "IOT2050 SM") {
++        RED.nodes.registerType('mraa-gpio-ain',{
++            category: 'GPIO',
++            color: '#a6bbcf',
++            paletteLabel: 'analogue',
++            defaults: {
++                name: {value:""},
++                pin:  {value:"", required: true, validate:RED.validators.number()},
++                interval: {value:"100", required: true, validate:RED.validators.number()}
++            },
++            inputs:0,
++            outputs:1,
++            icon: "arrow.png",
++            label: function() {
++                return this.name||"A"+this.pin;
++            },
++            oneditprepare: function() {
++                $('#node-input-pin').val(this.pin);
++                $('#btype').text(RED.settings.mraaGpioAinBoardType);
++                $('#ver-tip').text(RED.settings.mraaGpioAinMraaVersion);
++            }
++        });
++    }
+ </script>
+ 
+ <script type="text/html" data-template-name="mraa-gpio-ain">
+diff --git a/hardware/intel/mraa-gpio-din.html b/hardware/intel/mraa-gpio-din.html
+index 444bfa9c67eb..4906124f3bba 100644
+--- a/hardware/intel/mraa-gpio-din.html
++++ b/hardware/intel/mraa-gpio-din.html
+@@ -21,13 +21,23 @@
+             return this.name?"node_label_italic":"";
+         },
+         oneditprepare: function() {
+-            if (RED.settings.mraaGpioDinBoardType === "IOT2050") {
+-                for (i of [...Array(20).keys()].slice(14)) {
+-                    // from D14 to D19
++            if (RED.settings.mraaGpioDinBoardType === "IOT2050 SM") {
++                $('#node-input-pin').append($("<option></option>").attr("value", 0).text("USER button"));
++            } else {
++                for (i of [...Array(14).keys()]) {
++                    // from D0 to D13
+                     $('#node-input-pin').append($("<option></option>").attr("value", i).text("D" + i));
+                 }
+-                $('#node-input-pin').append($("<option></option>").attr("value", 20).text("USER button"));
++
++                if (RED.settings.mraaGpioDinBoardType === "IOT2050") {
++                    for (i of [...Array(20).keys()].slice(14)) {
++                        // from D14 to D19
++                        $('#node-input-pin').append($("<option></option>").attr("value", i).text("D" + i));
++                    }
```

Comment:
If you make the loop boundaries model-dependent, you can save one loop instance.

Reply:
Ack